### PR TITLE
fix: enable defaulter and validator in portal webhook

### LINKED ISSUE #264: failed to create portals with webhook enabled
This bug reported by this action: https://github.com/kubebb/core/actions/runs/5885812184/job/15962826594?pr=259

with error log

```text
7 try to verify portal works correctly
7.1 create two portals with different path and entry
Error from server (InternalError): error when creating "config/samples/core_v1alpha1_portal.yaml": Internal error occurred: failed calling webhook "vportal.kb.io": failed to call webhook: the server could not find the requested resource
```

### PR DESCRIPTION
**Pull request by bjwswang**
#### What type of PR is this?
/kind bug

#### Which issue(s) this PR fixes

Fixes #264 

#### Special notes for your reviewer

This webhook acutually has no specific functionalities yet,but it does sometime in the future.

### CODE REVIEW

**Comment by bjwswang** [comment]
@Henry-Gao-2004 After this pr got merged, you can continue your work.

File: api/v1alpha1/portal_webhook.go [hunk 1]
```diff
@@ -28,36 +28,48 @@ import (
...
+
+var _ webhook.CustomDefaulter = &Portal{}
+
+func (p *Portal) Default(ctx context.Context, obj runtime.Object) error {
+	portallog.Info("default", "name", p.Name)
+
+	return nil
+}
+
 //+kubebuilder:webhook:path=/validate-core-kubebb-k8s-com-cn-v1alpha1-portal,mutating=false,failurePolicy=fail,sideEffects=None,groups=core.kubebb.k8s.com.cn,resources=portals,verbs=create;update,versions=v1alpha1,name=vportal.kb.io,admissionReviewVersions=v1
```
Comment (Abirdcfly) [inline api/v1alpha1/portal_webhook.go:49]:
```suggestion
//+kubebuilder:webhook:path=/validate-core-kubebb-k8s-com-cn-v1alpha1-portal,mutating=false,failurePolicy=fail,sideEffects=None,groups=core.kubebb.k8s.com.cn,resources=portals,verbs=create;update;delete,versions=v1alpha1,name=vportal.kb.io,admissionReviewVersions=v1
```

If we want use Portal.ValidateDelete() and need run `make generate && make manifests`

Comment (bjwswang) [reply api/v1alpha1/portal_webhook.go:49]: Good point! `Delete` should also be vallidated.

**Review by Abirdcfly — APPROVED** [review]

**Comment by Henry-Gao-2004** [comment]
> @Henry-Gao-2004 After this pr got merged, you can continue your work.

Thank you! Nice work!

**Review by 0xff-dev — APPROVED** [review]

**Pull request merged**